map_gen/Diggy/Feature/NightTime.lua

```diff
@@ -0,0 +1,46 @@
+--[[-- info
+    Provides the ability to inform players that solar panels doesn't work underground
+]]
+
+-- dependencies
+local Event = require 'utils.event'
+local Game = require 'utils.game'
+
+-- this
+local NightTime = {}
+
+local function on_built_entity(event)
+    local player = Game.get_player_by_index(event.player_index)
+    local entity = event.created_entity
+    if (entity.name == 'solar-panel') then
+        require 'popup'.player(
+            player,[[
+Placing solar panels underground does not seem
+to have an effect on power production!
+Studies show, that the same applies to the portable version!
+
+Foreman's advice: Solar Panels are only useful in crafting
+satellites
+]]
+        )
+    end
+end
+
+local function on_research_finished(event)
+    local force = game.forces.player
```

Comment:
You can also do this to keep it scoped:
```suggestion
    local force = event.research.force
```

https://lua-api.factorio.com/latest/events.html#on_research_finished

Reply:
Great! I wondered if I could do that, but forgot to check.